### PR TITLE
Fix regression breaking countries and funds selection in quantiles.py

### LINKED ISSUE #27: `get_regions` no longer handles `countries` and `funds`
PR #26 includes a change to the logic in `get_regions`. There are a few problems with this change:
 - If the regions configuration option includes `countries` or `funds`, these will not be properly handled.
 - The function will return `[]` to mean "all regions" if `allregions` is None, and will return a list of all of the regions if it is given to it. That seems like confusing logic, when one of the points of `get_regions` is to abstract away the configuration setup and return a consistently usable result.

Probably the easiest solution is to remove the changes to `get_regions`, and move the call to `get_regions` to inside of `read_region`, at which point the regions list will be available. 

Also the line `if target_regions == ['all'] or target_regions == []:` should be replaced with a call to `is_allregions`.

### PR DESCRIPTION
**Pull request by brews**
Fix regression breaking countries and funds selection in quantiles.py, close #27

### CODE REVIEW

File: gcp/extract/lib/bundles.py [hunk 1]
```diff
@@ -48,14 +48,11 @@ def read_region(target_regions, *args, **kwargs):
     regions : array-like
     data : array-like
     """
-    # Because someone's always tryin to ice-skate up hill
-    if isinstance(target_regions, str):
-        target_regions = [target_regions]
-
     years, regions, data = read(*args, **kwargs)
+    target_regions = configs.get_regions(config, regions)
     regions_msk = np.isin(regions, target_regions)
```
Comment (jrising) [inline gcp/extract/lib/bundles.py]: Move this below the `configs.is_allregions` check, so we don't need to run an N^2 operation every time.

**Review by jrising — CHANGES_REQUESTED** [review]
This is good-- this was what I meant by my original suggestion. Just one change and it's ready.

**Comment by jrising** [comment]
Also, don't you need these lines that were previously in `get_regions`:
```
    if 'region' in config:
        return [config['region']]
```

File: gcp/extract/lib/bundles.py (hunk 1 above)
Comment (brews) [reply gcp/extract/lib/bundles.py]: Applied the recommended changes. Closing.

**Comment by brews** [comment]
- Added changes addressing your concerns mentioned in the review.
- Also change that correctly handles users using a `"global"` region in their run configs, as discussed in #27.

File: gcp/extract/lib/bundles.py [hunk 2]
```diff
@@ -48,15 +48,12 @@ def read_region(target_regions, *args, **kwargs):
...
-
     years, regions, data = read(*args, **kwargs)
-    regions_msk = np.isin(regions, target_regions)
+    regions_msk = np.ones(regions.shape, dtype='bool')
 
-    if target_regions == ['all'] or target_regions == []:
-        regions_msk[:] = True
+    if not configs.is_allregions(config):
+        target_regions = configs.get_regions(config, regions)
+        regions_msk[:] = np.isin(regions, target_regions)
```
Comment (jrising) [inline gcp/extract/lib/bundles.py]: Why aren't you just saying here:
```
if not configs.is_allregions(config):
    target_regions = configs.get_regions(config, regions)
    regions_msk = np.isin(regions, target_regions)
else:
    regions_msk = np.ones(regions.shape, dtype='bool')
```
? It seems weird to set up the variable just to replace it.

**Review by jrising — APPROVED** [review]
One stylistic difference, but good to go.

File: gcp/extract/lib/bundles.py (hunk 2 above)
Comment (brews) [reply gcp/extract/lib/bundles.py]: Good suggestion. I can also flip the if/else and drop the `not` to make it more clear. Just pushed a commit with this change. Thanks.

**Comment by jrising** [comment]
Thanks!

**Pull request merged**